agent/rpcclient/health/health.go

```diff
@@ -54,12 +52,7 @@ func (c *Client) getServiceNodes(
 		return out, cache.ResultMeta{}, err
 	}
 
-	cacheName := c.CacheName
-	if req.Connect {
-		cacheName = c.CacheNameConnect
-	}
-
-	raw, md, err := c.Cache.Get(ctx, cacheName, &req)
```

Comment:
Hmm I also think we still need this for the same reason. Clients that hit the `/v1/health/services` endpoint will still come through here but with `req.Connect = false` while `/v1/connect/services` will also come through here with `req.Connect = true`. We still need to switch Cache type based on which type of request is being made.